mslib/utils/_tests/test_airdata.py

```diff
@@ -0,0 +1,213 @@
+    assert path is not None
+    assert url is not None
+    text = '''<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+ <!-- For Testing ONLY -->
+<OPENAIP VERSION="1668386405436" DATAFORMAT="1.1" xmlns="https://www.openaip.net"\
+ xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"\
+  xsi:schemaLocation="https://www.openaip.net https://storage.googleapis.com/d644c13a-ed49-48ad-8493-d3e55d3281a5\
+  /assets/common/openaip-schema-v1.xsd">
+<AIRSPACES></AIRSPACES>
+</OPENAIP>
+'''
+    file_path = os.path.join(ROOT_DIR, "bg_asp.xml")
+    with open(file_path, "w") as f:
+        f.write(text)
+
+
+def _cleanup_test_files():
+    file_path = os.path.join(ROOT_DIR, "bg_asp.xml")
+    if "tmp" in file_path:
+        if os.path.exists(file_path):
+            os.remove(file_path)
+    file_path = os.path.join(ROOT_DIR, "airports.csv")
+    if "tmp" in file_path:
+        if os.path.exists(file_path):
+            os.remove(file_path)
+
+
+def test_download_progress():
+    file_path = os.path.join(ROOT_DIR, "airdata")
+    download_progress(file_path, 'http://speedtest.ftp.otenet.gr/files/test100k.db')
+    assert os.path.exists(file_path)
+
+
+@mock.patch("PyQt5.QtWidgets.QMessageBox.question", return_value=QtWidgets.QMessageBox.No)
+def test_get_airports(mockbox):
+    _cleanup_test_files()
+    airports = get_airports()
+    assert airports == []
+
+
+@mock.patch("PyQt5.QtWidgets.QMessageBox.question", return_value=QtWidgets.QMessageBox.Yes)
+def test_get_downloaded_airports(mockbox):
+    with mock.patch("mslib.utils.airdata.download_progress", _download_progress_airports):
+        airports = get_airports(force_download=True)
+        assert len(airports) > 0
+        assert 'continent' in airports[0].keys()
+        assert mockbox.critical.call_count == 0
+
+
+def test_get_available_airspaces():
+    _cleanup_test_files()
+    airspaces = get_available_airspaces()
+    assert len(airspaces) > 0
+
+
+@mock.patch("PyQt5.QtWidgets.QMessageBox.question", return_value=QtWidgets.QMessageBox.Yes)
+def test_update_airspace(mockbox):
+    with mock.patch("mslib.utils.airdata.download_progress", _download_progress_airspace):
+        update_airspace(force_download=True, countries=["bg"])
+        example_file = os.path.join(ROOT_DIR, "bg_asp.xml")
+        os.path.exists(example_file)
+        with open(example_file, 'r') as f:
+            text = f.read()
+        assert "<!-- For Testing ONLY -->" in text
+        assert mockbox.critical.call_count == 0
+
+
+@mock.patch("PyQt5.QtWidgets.QMessageBox.question", return_value=QtWidgets.QMessageBox.No)
+def test_get_airspaces_no_data(mockbox):
+    """
+    In the test environment we start always in a fresh tmp dir, no data is available
+    once it is downloaded it is managed by airdata
+    """
+    _cleanup_test_files()
+    airspaces = get_airspaces(countries=["bg"])
+    assert airspaces == []
+
+
+@mock.patch("mslib.utils.airdata.download_progress", _download_progress_airspace)
```

Comment:
OK, this particular use makes using a fixture more difficult, *unless* the mocking would be part of the fixture.
Using a fixture had the added benefit that one could "yield" the correct data, ideally using the same data structure for yielding and generating the xml file.

Reply:
Currently we see the tests describing the existing code.  That this is needed shows a complexity we should get refactor in develop.